bundles/org.openhab.binding.tacmi/src/main/java/org/openhab/binding/tacmi/internal/TACmiHandler.java

```diff
@@ -0,0 +1,337 @@
+
+        if (config.host == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "No host configured!");
+            return;
+        }
+        this.online = false;
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING);
+
+        // set cmiAddress from configuration
+        // cmiAddress = (String) configuration.get("cmiAddress");
+        try {
+            cmiAddress = InetAddress.getByName(config.host);
+        } catch (final UnknownHostException e1) {
+            logger.error("Failed to get IP of CMI from configuration");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Failed to get IP of CMI from configuration");
+            return;
+        }
+
+        this.node = config.node;
+
+        // initialize lookup maps...
+        this.channelConfigByUID.clear();
+        for (final Channel chann : getThing().getChannels()) {
+            try {
+                final ChannelTypeUID ct = chann.getChannelTypeUID();
+                final boolean analog = CHANNEL_TYPE_COE_ANALOG_IN_UID.equals(ct)
+                        || CHANNEL_TYPE_COE_ANALOG_OUT_UID.equals(ct);
+                final boolean outgoing = CHANNEL_TYPE_COE_ANALOG_OUT_UID.equals(ct)
+                        || CHANNEL_TYPE_COE_DIGITAL_OUT_UID.equals(ct);
+                // for the analog out channel we have the measurement type. for the input
+                // channel we take it from the C.M.I.
+                final Class<? extends TACmiChannelConfiguration> ccClass = CHANNEL_TYPE_COE_ANALOG_OUT_UID.equals(ct)
+                        ? TACmiChannelConfigurationAnalog.class
+                        : TACmiChannelConfiguration.class;
+                final TACmiChannelConfiguration cc = chann.getConfiguration().as(ccClass);
+                this.channelConfigByUID.put(chann.getUID(), cc);
+                final MessageType messageType = analog ? MessageType.A : MessageType.D;
+                final byte podId = this.getPodId(messageType, cc.output);
+                final PodIdentifier pi = new PodIdentifier(messageType, podId, outgoing);
+                // initialzie podData
+                getPodData(pi);
+            } catch (final Exception e) {
+                logger.error("Failed to collect data for Channel {}: {}", chann.getUID(), e.getMessage(), e);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "Failed to collect data for Channel " + chann.getUID() + ": " + e.getMessage());
+                return;
+            }
+
+        }
+
+        // this automatically restores persisted states...
+        this.stateCacheUtils = new StateCacheUtils(
+                new File(STATE_CACHE_BASE + getThing().getUID().getAsString().replace(':', '_') + ".json"),
+                this.podDatas.values());
```

Comment:
I tried to solve a 'startup' gap between the two architectures of OH and the TA System: TA transfers always a set of 4 or 16 channel states (depending weather we have analog values or switch states) in one message, so the binding has to have the states of the 'neighboring' channels available when one channel gets updated. Is there an other way to restore / read 'previous' values during startup?

Reply:
I'm not aware any binding stores the state of its Things persistently on disk. Isn't the state of the Things outdated if OH has been switched off for some time?

The normal way is to react to a RefreshType command. This will be issued on startup by the OH core to the binding, to signal the binding it should poll the current state from its Things.

Reply:
Hi,

Okay, it might be good to explain my whole user-story to get a better idea why i persisted the data.
First of all, we have two flow directions for data: One that delivers data from TA -> OH, and one that sends data from OH to TA.
In my scenario I need the direction OH -> TA:
I have some old 'legacy' hardware/equipment in my heating system that is connected to OH. So OH knows states that need to be delivered to the TA system, as the TA hast to work with it. Also the old 'legacy' stuff doesn't have possibility to poll states, I have to wait for them to be updated. There are values that are updated very frequently (on a per-second basis) but others are only updated every 15 minutes.
In OH I've configured MAPDB as persistence service, so after a restart it comes up with the last values. That's fine.
In OH there are also rules for each of the required 'legacy' item that whenever it gets updated/changed, the new value is sent to the TA output channel. This also works as expected.

The problem is the way the values have to be transferred to the TA:
Every time a value is transferred to the TA, there package will include at least 4 values. So I also have to send the current / last values from the 3 neighboring channels. The plugin saves the last values from the neighboring things, as long as there is a value.
And now the problem is: the incoming values have very different update intervals. Some values are updated quite often, others not - but when I want to send an Update to TA for the frequent channel I also have to send the values of the non-so-frequent values. So I have to find a solution to get the current / last values of these neighboring channels. Even if the last known value is a bit older it usually is much better then sending just 0 to the TA is it would cause totally wrong assumptions / calculations on the TA site.

So what I need here during startup of the plugin is to find a way to get some useful initial state - either the last values known (why i'm currently persisting it) or to read the current value of the linked 'item' from OH. Like the RefreshType, but the other way around - the binding sens a RefreshType to the OH Core to get the last known value from the linked Item. Is there such a mechanism available? Or any other hint?

Reply:
I understood that legacy values cannot be polled and you cannot set single values, but must update several values at once. And you don't know the states of the other values after OH started.

I'm sorry for this design and I understand why you chose persisting the values. Unfortunately I don't have any better idea at the moment...

In the last paragraph, I guess you refer to using mapdb. Honestly I'm not sure if you can get access from within a binding to mapdb. If so, that would be a nicer way, instead introducing a custom persistence. However, it might be a good idea to continue this discussion in the development forum.